File: 프로그래머스/Level 2/k진수에서 소수 개수 구하기-ex.java
class Solution {
    public int solution(int n, int k) {
        int answer = 0;
        String temp[] = Integer.toString(n, k).split("0");
        
        Loop : for (String t : temp) {
            if (t.length() == 0) continue;
            long a = Long.parseLong(t);
            if (a == 1) continue;
            for (int i = 2; i <= Math.sqrt(a); i++) 
                if (a % i == 0) continue Loop;
            
            answer++;
        }
        
        return answer;
    }
}

File: 프로그래머스/Level 2/k진수에서 소수 개수 구하기-ex.py
def conv(n, k):
    s = ''
    while n:
        s += str(n % k)
        n //= k
    
    return s[::-1]

def isprime(n):
    if n <= 1: return False 
    i = 2
    while i * i <= n:
        if n % i == 0: return False
        i += 1
    return True

def solution(n, k):
    s = conv(n, k)
    cnt = 0
    for num in s.split('0'):
        if not num: continue
        if isprime(int(num)): cnt += 1
    return cnt